### PR TITLE
Fixes #1049. Upload images before form submission.

### LINKED ISSUE #1049: Automatic screenshot uploads
I think we should move to a more GitHub-like image upload solution:

1) User selects screenshot (or has one from the add-on)
2) We automatically upload that and insert the URL to the image in the `<textarea>` while the user is filling out the rest of the bug details
3) ...
4) Profit!

### PR DESCRIPTION
**Pull request by miketaylr**

### CODE REVIEW

**Comment by miketaylr** [comment]
Ah yeah, need to update python tests.

**Comment by miketaylr** [comment]
OK! Just deployed this to staging if anyone wants to test.

So far, my tests are good! Basically, reporting an issue with a screenshot as an authed GitHub user works again. And it's soooo much faster thanks to @karlcow.

r? @karlcow

**Comment by karlcow** [comment]
This is working.

```
127.0.0.1 - - [20/May/2016 11:04:32] "POST /issues/new HTTP/1.1" 302 -
127.0.0.1 - - [20/May/2016 11:04:32] "GET /login HTTP/1.1" 302 -
127.0.0.1 - - [20/May/2016 11:04:42] "GET /callback?code=******* HTTP/1.1" 302 -
127.0.0.1 - - [20/May/2016 11:04:44] "GET /file HTTP/1.1" 302 -
127.0.0.1 - - [20/May/2016 11:04:44] "GET /thanks/397 HTTP/1.1" 200 -
127.0.0.1 - - [20/May/2016 11:05:22] "GET /issues/397 HTTP/1.1" 200 -
127.0.0.1 - - [20/May/2016 11:05:22] "GET /js/vendor/qr.min.js HTTP/1.1" 200 -
127.0.0.1 - - [20/May/2016 11:05:22] "GET /js/lib/models/comment.js HTTP/1.1" 200 -
127.0.0.1 - - [20/May/2016 11:05:22] "GET /js/lib/issues.js HTTP/1.1" 200 -
127.0.0.1 - - [20/May/2016 11:05:22] "GET /js/lib/qrcode.js HTTP/1.1" 200 -
127.0.0.1 - - [20/May/2016 11:05:22] "GET /js/lib/comments.js HTTP/1.1" 200 -
127.0.0.1 - - [20/May/2016 11:05:24] "GET /api/issues/labels?per_page=100 HTTP/1.1" 200 -
127.0.0.1 - - [20/May/2016 11:05:24] "GET /api/issues/397 HTTP/1.1" 200 -
127.0.0.1 - - [20/May/2016 11:05:24] "GET /uploads/2016/5/c701a0ad-0652-4325-b237-bf072a08b675.jpg HTTP/1.1" 200 -
```

File: webcompat/static/js/lib/bugform.js [hunk 1]
```diff
@@ -347,17 +324,70 @@ function BugForm() {
...
       label.css('background', 'none');
       removeBanner.addClass('is-hidden');
       uploadWrapper.removeClass('is-hidden');
       removeBanner.off('click');
+
+      // remove the last embedded image URL
+      // Note: this could fail in weird ways depending on how
+      // the user has edited the descField.
+      this.descField.val(function(idx, value) {
+        return value.replace(/!\[.+\.jpe*g\)$/, '');
```
Comment (karlcow) [inline webcompat/static/js/lib/bugform.js]: I don't get this regex here. 
What is it supposed to do.

**Comment by karlcow** [comment]
r+ @miketaylr

**Comment by karlcow** [comment]
Super extra with the two latest pull requests I have the feeling with suddenly bumped our UX ;) or at least unbroke it.

**Comment by miketaylr** [comment]
> I have the feeling with suddenly bumped our UX ;) or at least unbroke it.

Little bit from Column A, little bit from Column B. :p

File: webcompat/static/js/lib/bugform.js (hunk 1 above)
Comment (miketaylr) [inline webcompat/static/js/lib/bugform.js]: When you click "Remove image upload", it removes the image preview *and* removes the last `Screenshot descriptions` in the bug description. And actually it's got a bug in it! We can also upload gif and bmp (png will always be jpg).

Comment (karlcow) [inline webcompat/static/js/lib/bugform.js]: capture d ecran 2016-05-21 a 06 41 11

**Pull request merged**